lib/src/main/java/com/dynatrace/metric/util/Normalizer.java

```diff
@@ -257,6 +258,139 @@ static NormalizationResult normalizeDimensionValue(String value, int maxDimensio
     return normalizeUnquotedStringDimValue(value, maxDimensionValueLength);
   }
 
+  static NormalizationResult normalizeMetadataString(String value, int maxLength) {
+    if (StringValueValidator.isNullOrEmpty(value)) {
+      return NormalizationResult.newValid(CodePoints.EMPTY_STRING);
+    }
+
+    // in a quoted string, quotes do not count towards the string character limit
+    boolean isQuoted =
+        value.startsWith(CodePoints.QUOTATION_MARK) && value.endsWith(CodePoints.QUOTATION_MARK);
+
+    int numValidBytes = 0;
+    boolean needsNormalization = false;
+    boolean needsEscaping = false;
+
+    // if quoted, start at index 1 (the one after the quote)
+    int start = isQuoted ? 1 : 0;
+    // if quoted, stop at length - 1 (the last byte before the quote)
+    int end = isQuoted ? value.length() - 1 : value.length();
+
+    for (int i = start; i < end; ) {
+      final int codePoint = value.codePointAt(i);
+      final int codePointLength = Character.charCount(codePoint);
+
+      // code point is valid but needs escaping
+      if (codePointNeedsEscaping(codePoint)) {
+        needsEscaping = true;
+      }
+
+      // code point is not valid, and needs normalizing (e.g. unicode control char)
+      if (codePointNeedsNormalizing(codePoint)) {
+        // if there are control characters, note that normalization is necessary but don't break yet
+        // this way we know how many characters there will be in the output
+        needsNormalization = true;
+      } else {
+        // if the char is valid, count up
+        numValidBytes += codePointLength;
+        // if the maxLength is reached, break.
+        if (numValidBytes > maxLength) {
+          needsNormalization = true;
+          break;
+        }
+      }
+      i += codePointLength;
+    }
+
+    // return early if no normalization is needed.
+    if (!needsNormalization) {
+      // if the string is already quoted or doesn't need escaping, return as is
+      if (isQuoted || !needsEscaping) {
+        return NormalizationResult.newValid(value);
+      } else {
+        // not quoted, needs escaping, but no normalization (e.g., it contains a '=' character).
+        // turn the string into a quoted string, where the '=' sign does not need escaping.
+        return NormalizationResult.newValid("\"" + value + "\"");
+      }
+    }
+
+    // do normalization
+    StringBuilder builder = new StringBuilder(numValidBytes + 2);
+    builder.append(CodePoints.QUOTATION_MARK);
+    Supplier<String> warningMessageSupplier = null;
+
+    // don't need to escape anything since were working in a quoted string.
```

Comment:
```suggestion
    // don't need to escape anything since we're working in a quoted string.
```
Can we always do this though? Doesn't the quote chars also count towards the max limit?

Reply:
No, the enclosing quotes in a quoted string do not count towards the limit.